client/src/api/tasks.js

```diff
@@ -14,16 +14,25 @@ export async function getAllTasks() {
     throw new Error(response.statusText);
   }
   const tasks = await response.json();
-
   return tasks;
 }
 
-// export async function editTask(taskId) {
-//   const response = await fetch(`/tasks/${taskId}`, { method: 'GET' });
-//   if (!response.ok) {
-//     throw new Error(response.statusText);
-//   }
-//   const task = await response.json();
+export async function patchTask(
+  taskId,
+  date,
+  startTime,
+  endTime,
+  category,
+  heading
+) {
```

Comment:
If you have multiple arguments, it makes sense to reduce it to a single argument and destruct it:

```suggestion
export async function patchTask({
  taskId,
  date,
  startTime,
  endTime,
  category,
  heading
}) {
```

the advantage is, that you don't have to take care about the order of the arguments.

```js
patchTask({
  taskId: 1,
  startTime: 123123,
  endTime: 123124,
  heading: 'abc',
  date: new Date(),
  category: "c"
});
```

But in your case, it would be even smarter to have something like this:

```suggestion
export async function patchTask(
  taskId,
  fields
) {
```

You can bypass `fields` to your JSON.stringify and don't have to update this function if you want to add or remove fields.

Reply:
oke habe ich versucht aber dann bekomme ich nur Fehlermeldungen und mich damit auseinanderzusetzen hab ich grad die Zeit nicht. Ich versuche grad alles soweit fertig zubekommen dass ich das Video machen kann 🙈